bower.json

```diff
@@ -15,7 +15,11 @@
     "purescript-prelude": "^3.0.0",
     "purescript-console": "^3.0.0",
     "purescript-integers": "^3.0.0",
-    "purescript-st": "^3.0.0",
-    "purescript-partial": "^1.2.0"
+    "purescript-partial": "^1.2.0",
+    "purescript-arrays": "^4.2.1",
+    "purescript-foldable-traversable": "^3.6.1"
+  },
+  "devDependencies": {
+    "purescript-node-fs": "^4.0.0"
```

Comment:
I think it would be good to test this without the `node-fs` dependency. Perhaps we can use a foreign module for this one function. The reason is that dependencies like this one make it harder to update libraries later, since we have to wait until a lot more libraries are updated downstream until we can come back and finish off the core libraries.

Reply:
I ended up simplifying this quite a bit and just made a version of the `loop` function that calls a callback on completion, which is sufficient for testing.